### PR TITLE
build: make podman_build a function

### LINKED ISSUE #111: `build.sh` will fail if the path containing it has spaces
Every space separates the path into new args.

I believe many people looking for a reliable way to build Godot binaries will stop by this repo and may not know this.

I'm thinking about adding a notice about the path.

### PR DESCRIPTION
**Pull request by umarcor**
This PR is based on #119.
It should fix #111 and superseed #112.

### CODE REVIEW

File: build.sh [hunk 1]
```diff
@@ -43,36 +43,51 @@ done
...
-$podman build -t godot-fedora:${img_version} -f Dockerfile.base . 2>&1 | tee logs/base.log
-$podman_build -t godot-export:${img_version} -f Dockerfile.export . 2>&1 | tee logs/export.log
+podman_build() {
+  # You can add --no-cache  as an option to podman_build below to rebuild all containers from  scratch.
+  "$podman" build \
+  --build-arg img_version=${img_version} \
+  -v "${files_root}":/root/files:z \
+  -t godot-"$1:${img_version}" \
+  -f Dockerfile."$1" . \
+  2>&1 | tee logs/"$1".log
```
Comment (akien-mga) [inline build.sh:55]:
```suggestion
  "$podman" build \
    --build-arg img_version=${img_version} \
    -v "${files_root}":/root/files:z \
    -t godot-"$1:${img_version}" \
    -f Dockerfile."$1" . \
    2>&1 | tee logs/"$1".log
```

File: build.sh [hunk 2]
```diff
@@ -43,36 +43,51 @@ done
...
+  if [ ! -e "${files_root}"/Xcode_${XCODE_SDK}.xip ]; then
     echo "files/Xcode_${XCODE_SDK}.xip is required. It can be downloaded from https://developer.apple.com/download/more/ with a valid apple ID."
     exit 1
   fi
 
   echo "Building OSX and iOS SDK packages. This will take a while"
-  $podman_build -t godot-xcode-packer:${img_version} -f Dockerfile.xcode . 2>&1 | tee logs/xcode.log
-  $podman run -it --rm -v ${files_root}:/root/files:z -e XCODE_SDKV="${XCODE_SDK}" -e OSX_SDKV="${OSX_SDK}" -e IOS_SDKV="${IOS_SDK}" godot-xcode-packer:${img_version} 2>&1 | tee logs/xcode_packer.log
+  podman_build xcode
+  $podman run -it --rm \
```
Comment (akien-mga) [inline build.sh:77]: You changed other uses of `$podman` for `"$podman"` but missed this one.

**Comment by umarcor** [comment]
Since #119 was merged, this is now ready for merging!

**Comment by umarcor** [comment]
@akien-mga @Calinou can this be merged?

**Comment by akien-mga** [comment]
Thanks!

**Pull request merged**